packages/common/src/constraints/CheckCandidates.ts

```diff
@@ -142,7 +149,8 @@ export const checkCandidate = async (
       blockedValid &&
       kusamaValid &&
       providerValid &&
-      beefyValid;
+      beefyValid &&
+      locationValid;
```

Comment:
location as a name is misleading in this context: location unfortunately encompass more than just country and region (the things that you are interested to filter), i.e. provider, cpu, ... 

thus the fact that you are checking locationValid is not true (i.e. provider is checked 2 lines above)

I suggest a name such as `geoAreaValid` or similar, and if this is targeting just the sanctions I'd go even further with `geoAreaSanctionsValid`